src/sql/parts/connection/connectionDialog/connectionWidget.ts

```diff
@@ -120,17 +128,35 @@ export class ConnectionWidget {
 		if (this._authTypeSelectBox) {
 			this.onAuthTypeSelected(this._authTypeSelectBox.value);
 		}
+
+		DOM.addDisposableListener(container, 'paste', e => {
+			this._handleClipboard();
+		});
+
 		DOM.append(container, this._builder.getHTMLElement());
 	}
 
+	private _handleClipboard(): void {
+		let paste = this._clipboardService.readText();
+		this._connectionManagementService.serializeConnectionString(paste, this._providerName).then(e => {
+			if (e) {
```

Comment:
If the user pastes something that's not a connection string what happens? We should make sure an error doesn't show up in the debug console or anything like that

Reply:
It will return undefined, and therefore nothing will happen